pkg/kubelet/network/dns/dns.go

```diff
@@ -230,7 +230,12 @@ func parseResolvConf(reader io.Reader) (nameservers []string, searches []string,
 			}
 		}
 		if fields[0] == "search" {
-			searches = fields[1:]
+			// Strip redundant trailing dot to avoid hitting search validation limits.
+			trimTrailingDot := []string{}
```

Comment:
why can't use just use `searches` here?

Reply:
Well, I might be misunderstanding your proposal, but if it looks like this:

```go
if fields[0] == "search" {
    for _, s := range fields[1:] {
        searches = append(searches, strings.TrimSuffix(s, "."))
   }
}
```
This would not work with an 2 lines searches:

https://play.golang.org/p/T3znpev_G1E

This actually breaks https://github.com/kubernetes/kubernetes/blob/0b850c5d82389fff3cf241127c9bd80bb03198f6/pkg/kubelet/network/dns/dns_test.go#L79

Does that make sense to you?

Reply:
I think he just meant use the existing variable instead of creating an additional one, i.e.:
```
		if fields[0] == "search" {
			searches = []string{}
			for _, s := range fields[1:] {
				searches = append(searches, strings.TrimSuffix(s, "."))
			}
		}
```

Reply:
ah right, ok, lgtm too :). I'll push it.

Reply:
Thanks @bboreham !!

Reply:
yup, that's what I meant